emcc.py

```diff
@@ -383,6 +383,12 @@ def get_clang_flags(user_args):
       flags.append('-matomics')
     if '-mbulk-memory' not in user_args:
       flags.append('-mbulk-memory')
+  elif '-mbulk-memory' not in user_args and '-mno-bulk-memory' not in user_args:
```

Comment:
Perhaps a separate block here rather then trying chain with the above if/else, along with a comment?

How about:

```
# In emscripten we currently disable bulk memory by default
if not settings.BULK_MEMORY and '-mno-bulk-memory' not in user_args:
  flags.append('-mno-bulk-memory')
```

The `not settings.BULK_MEMORY` already encompasses the pthread and shared memory and the explict `-bulk-memory` cases, which all enable this setting.